### DISCUSSION TITLE
HELP (Anyone can help me with this problem)

### DISCUSSION THREAD

**Discussion by urkoperez16** [Sehelitar/Kick.bot#57] [Q&A]
Its my first time using Streamer.bot and I don't know how it works and I don't know how to integrate this plugin for kick too.
if someone could help me I would be very grateful (sorry for my bad English)

**Comment by Sehelitar** [comment]
Hi! 👋 😊
If you need help with Streamer.bot itself, you can take a look at the documentation here : https://docs.streamer.bot/get-started/introduction
They have some video tutorials on Youtube and a Discord server if you need more help.
As for Kick.bot, the documentation is available here : https://github.com/Sehelitar/Kick.bot/wiki/Installation
It explains the installation process but requires that you first know how Streamer.bot works.

**Comment by urkoperez16** [comment]
Thank you very much for replying, I have seen the youtube video on discord and everything is clear.

**Comment by urkoperez16** [comment]
Now I have another question. I'm configuring Streamer.bot for a multiplatform subathon, I want each sub to send a message to twitch to add time (I have it all configured by twitch) so, each normal sub sends a message to twitch, but when they give away more than 1 sub, I don't know how to make it send the same amount of messages. At the moment I have this part:
![image](https://github.com/Sehelitar/Kick.bot/assets/105135600/ebb9fae8-c27c-4301-83a0-227dfdcb4ba4)

**Reply by Sehelitar** [reply]
You have to create a loop to send the message X times, depending on how many subs has been gifted.
It should looks like this :
![image](https://github.com/Sehelitar/Kick.bot/assets/1716704/7b2fb785-515b-4cf7-bca3-22abf01797fa)
What it does is :
1. Send a !sub command to Twitch
2. Decrement the gifts counter (which acts as a loop counter) by setting ``gifts`` argument to ``$math(%gifts% - 1)$``
3. If ``gifts`` is still > 0, call that action again to loop. If ``gifts`` > 0, call your Kick gifts action then break, else just do nothing.

**Reply by urkoperez16** [reply]
So, the action keeps repeating until the gifted subscriptions counter is 0?

**Reply by Sehelitar** [reply]
That's the idea, yes. 👍

**Comment by urkoperez16** [comment]
I have another doubt, the streamer wants to save the users who subscribe or give away subscriptions for kick.  If they subscribe per kick, save their id 4 times, if they give away 10 subs save their id 4x10 times. do you know if it is possible?

**Reply by Sehelitar** [reply]
It should be possible, yes. If you know how to code in C#, Streamer.bot can do A LOT of things. And if you don't, some things might be a bit tricky to put in place, but ultimately, you can always find a way.
